lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -3065,26 +3057,47 @@ static bool isImporterGeneratedAccessor(const clang::Decl *clangDecl,
 
 static CanSILFunctionType getUncachedSILFunctionTypeForConstant(
     TypeConverter &TC, TypeExpansionContext context, SILDeclRef constant,
-    CanAnyFunctionType origLoweredInterfaceType) {
-  assert(origLoweredInterfaceType->getExtInfo().getSILRepresentation()
-           != SILFunctionTypeRepresentation::Thick
-         && origLoweredInterfaceType->getExtInfo().getSILRepresentation()
-             != SILFunctionTypeRepresentation::Block);
+    TypeConverter::LoweredFormalTypes bridgedTypes) {
+  auto silRep = TC.getDeclRefRepresentation(constant);
+  assert(silRep != SILFunctionTypeRepresentation::Thick &&
+         silRep != SILFunctionTypeRepresentation::Block);
 
+  auto origLoweredInterfaceType = bridgedTypes.Uncurried;
   auto extInfo = origLoweredInterfaceType->getExtInfo();
 
+  auto extInfoBuilder =
+      SILExtInfo(extInfo, false).intoBuilder().withRepresentation(silRep);
+
+  if (shouldStoreClangType(silRep)) {
+    const clang::Type *clangType = nullptr;
+    if (bridgedTypes.Pattern.isClangType()) {
+      clangType = bridgedTypes.Pattern.getClangType();
+    }
+    if (clangType) {
+      // According to [NOTE: ClangTypeInfo-contents], we need to wrap a function
+      // type in an additional clang::PointerType.
```

Comment:
What is this NOTE markup?  Is this an editor feature we're assuming support for, or is it just a reference to the documentation?

Reply:
It's a cross-reference to documentation elsewhere. It's not supported by any editors, it just helps to indicate coupling between two things. Generally, I prefer to document invariants etc. near the type definition/invariant check. So when you change the type definition/invariant, it is easier to track down what else is relying on it. Another benefit is that when you need to write a longer comment, you can write it out-of-line, which avoids obscuring the surrounding code. https://github.com/apple/swift/blob/master/lib/FrontendTool/FrontendTool.cpp has a bunch of similar named NOTEs because of the different edge cases in the module trace.